components/chartmogul/actions/search-customers-via-email/search-customers-via-email.mjs

```diff
@@ -19,19 +19,23 @@ export default {
   async run({ $ }) {
     const { email } = this;
 
-    const items = this.chartmogul.paginate({
-      $,
-      fn: this.chartmogul.searchCustomers,
-      params: {
-        email,
-      },
-    });
+    try {
+      const items = this.chartmogul.paginate({
+        $,
+        fn: this.chartmogul.searchCustomers,
+        params: {
+          email,
+        },
+      });
 
-    const response = [];
-    for await (const item of items) {
-      response.push(item);
+      const response = [];
+      for await (const item of items) {
+        response.push(item);
+      }
+      $.export("$summary", "Customers Successfully fetched");
+      return response.reverse();
+    } catch (e) {
+      return;
```

Comment:
you should expose the error for better use experience, in this case, you could log the original error `log(e);` and throw a ConfigurationError with a custom message.